### PR TITLE
CNV-10283: CDI customizations in HCO

### PR DESCRIPTION
**Pull request by ousleyp**
- [CNV-10283](https://issues.redhat.com/browse/CNV-10283)
- enterprise-4.8 only

Preview builds:
- [Preparing CDI scratch space ](https://deploy-preview-33658--osdocs.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virtual_disks/virt-preparing-cdi-scratch-space.html)
- [Configuring CDI to work with namespaces that have a compute resource quota](https://deploy-preview-33658--osdocs.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virtual_disks/virt-configuring-cdi-for-namespace-resourcequota?utm_source=github&utm_campaign=bot_dp)

### CODE REVIEW

File: modules/virt-about-cpu-and-memory-quota-namespace.adoc [hunk 1]
```diff
@@ -7,5 +7,5 @@
 
 A _resource quota_, defined by the `ResourceQuota` object, imposes restrictions on a namespace that limit the total amount of compute resources that can be consumed by resources within that namespace.
 
-The `CDI` object defines the user configuration for the Containerized Data Importer (CDI). The CPU and memory request and limit values for the `CDI` object are set to a default value of 0.
+The `HyperConverged` custom resource (CR) defines the user configuration for the Containerized Data Importer (CDI). The CPU and memory request and limit values are set to a default value of 0.
```
Comment (ousleyp) [inline modules/virt-about-cpu-and-memory-quota-namespace.adoc:10]: @aglitke Is this still accurate? Thanks!

**Review by nunnatsa — APPROVED** [review]
/lgtm

File: modules/virt-about-cpu-and-memory-quota-namespace.adoc (hunk 1 above)
Comment (aglitke) [reply modules/virt-about-cpu-and-memory-quota-namespace.adoc:10]: I think it is superfluous to name individual components that HCO configures.  The point of the feature is to concentrate all configuration in one place regardless of the underlying details.

File: modules/virt-defining-storageclass.adoc [hunk 2]
```diff
@@ -0,0 +1,37 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/virtual_disks/virt-preparing-cdi-scratch-space.adoc
+
+[id="virt-defining-storageclass_{context}"]
+= Defining a storage class
+
+Dynamically provision scratch space for CDI operations by defining a storage class in the `HyperConverged` custom resource (CR).
```
Comment (aglitke) [inline modules/virt-defining-storageclass.adoc:8]: I think we should change the wording here.  The configuration references an existing storage class which CDI will use when allocating scratch space for certain operations.

File: modules/virt-defining-storageclass.adoc [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+apiVersion: hco.kubevirt.io/v1beta1
+kind: HyperConverged
+metadata:
+  name: kubevirt-hyperconverged
+spec:
+  scratchSpaceStorageClass: "<storage_class>" <1>
+----
+<1> Replace `<storage_class>` with the storage class that you want to use to provision scratch space for CDI operations in your cluster.
+
+. Save and exit your default editor to update the `HyperConverged` CR.
```
Comment (aglitke) [inline modules/virt-defining-storageclass.adoc:37]: We may want to mention that when this value is not specified, scratch space will be allocated from the same storage class as the pvc that is being populated.

File: modules/virt-about-cpu-and-memory-quota-namespace.adoc (hunk 1 above)
Comment (ousleyp) [reply modules/virt-about-cpu-and-memory-quota-namespace.adoc:10]: You are probably right that we're giving too much detail there, @aglitke, and I'll take that into consideration. I meant to ask if the statement "The CPU memory request and limit values are set to a default value of 0" is still accurate. Thanks!

Comment (apinnick) [inline modules/virt-about-cpu-and-memory-quota-namespace.adoc:10]: Should the `0` value be in backticks?

File: modules/virt-defining-storageclass.adoc [hunk 4]
```diff
@@ -0,0 +1,36 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/virtual_disks/virt-preparing-cdi-scratch-space.adoc
+
+[id="virt-defining-storageclass_{context}"]
+= Defining a storage class
+
+To specify the storage class that the Containerized Data Importer (CDI) uses when allocating scratch space, add the `spec.scratchSpaceStorageClass` field to the `HyperConverged` custom resource (CR).
```
Comment (apinnick) [inline modules/virt-defining-storageclass.adoc:8]: I prefer the same verb in the header and introduction. So I might phrase it like this:

You can define the storage class that the CDI uses when allocating scratch space by adding the spec.etc parameter to the HyperConverged CR.

or 

To define the storage class ..., add the spec.scratchSC parameter to the CR.

File: modules/virt-defining-storageclass.adoc [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+[source,yaml]
+----
+apiVersion: hco.kubevirt.io/v1beta1
+kind: HyperConverged
+metadata:
+  name: kubevirt-hyperconverged
+spec:
+  scratchSpaceStorageClass: "<storage_class>" <1>
+----
+<1> Replace `<storage_class>` with the storage class that you want to use to provision scratch space for CDI operations in your cluster. If you do not specify a value, scratch space is allocated from the storage class used by the PersistentVolumeClaim (PVC) that is being populated.
```
Comment (apinnick) [inline modules/virt-defining-storageclass.adoc:34]: I am not sure you need the first sentence. The introduction and the name of the parameter make it clear that this is for scratch space for CDI. 

Suggestion: <1> If you do not specify a storage class, the CDI uses the storage class of the persistent volume claim that is being populated.

**Comment by apinnick** [comment]
LGTM. Small comments left.

**Pull request merged**